korgan/miniapp_api_v5.py

```
@@ -0,0 +1,467 @@
+    The Mini App API is a separate Railway process from the Telegram agent. A
+    per-user lock is held across each Mini App request, including long AI work,
+    so state load/mutate/save sequences for one identity cannot race each other.
+    """
+    if not request.url.path.startswith("/miniapp/"):
+        return await call_next(request)
+    raw = request.headers.get("x-telegram-init-data", "")
+    if not raw:
+        return await call_next(request)
+    try:
+        identity = core.legacy._identity(raw)
+    except Exception:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)